src/appleseed/renderer/kernel/intersection/assemblytree.cpp

```diff
@@ -77,9 +77,9 @@ namespace renderer
 // AssemblyTree class implementation.
 //
 
-AssemblyTree::AssemblyTree(const Scene& scene)
+AssemblyTree::AssemblyTree(const Project& project)
   : TreeType(AlignedAllocator<void>(System::get_l1_data_cache_line_size()))
-  , m_scene(scene)
+  , m_project(project)
```

Comment:
We just need the scene here. Why this change ?

Reply:
I think it's to TriangleTree the project so that it can find the texture store. If so, no need to change the members of AssemblyTree, just do `m_scene(scene)` or `m_scene(project.get_scene())`.

Reply:
Doesn't keeping both `Scene` and `Project` as members redundant? I mean `Scene` could be easily accessed from `m_project`, so why keep it?

Reply:
Storing a reference to the project in `AssemblyTree` is the right solution.